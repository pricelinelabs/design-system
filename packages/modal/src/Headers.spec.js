import React from 'react'
import { render } from 'testing-library'
import { ModalHeader, SmallModalHeader } from './index'

describe('SmallModalHeader', () => {
  test('render', () => {
    const { container } = render(<SmallModalHeader color='secondary' />)
    expect(container).toMatchSnapshot()
  })
})

describe('ModalHeader', () => {
  test('render', () => {
    const { container } = render(
      <ModalHeader
        color='secondary'
        title='test'
        onClose={() => {
          console.log('close')
        }}
      />
    )
    expect(container).toMatchSnapshot()
  })
})
