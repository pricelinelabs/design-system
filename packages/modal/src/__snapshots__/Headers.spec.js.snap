// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalHeader render 1`] = `
.c1 {
  padding-left: 16px;
  padding-right: 16px;
  background-color: #0a0;
  color: #fff;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  font-weight: 700;
  font-size: 14px;
}

.c9 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  width: 24px;
}

.c10 {
  outline: none;
}

.c6 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: inherit;
  font-weight: 700;
  line-height: 1.5;
  cursor: pointer;
  border-radius: 2px;
  border-width: 0;
  border-style: solid;
  background-color: #0068EF;
  color: #fff;
  margin-left: auto;
}

.c6:hover {
  background-color: #049;
  color: #fff;
}

.c6:focus {
  outline: 0px solid #049;
  box-shadow: 0 0 0 2px #049;
}

.c6:disabled {
  cursor: not-allowed;
  color: #4f6f8f;
  background-color: #edf0f3;
}

.c7 {
  padding: 0;
  background-color: transparent;
  color: inherit;
}

.c7:hover {
  background-color: transparent;
  color: inherit;
}

.c7 > div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7:disabled {
  background-color: transparent;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c3 {
  height: 40px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 {
  height: 24px;
  width: 24px;
}

.c8 svg {
  vertical-align: top;
}

.c8:focus {
  background-color: #049;
  outline: none;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1 c2 c3"
      color="secondary"
    >
      <div
        class="c4 c5"
        font-size="1"
      >
        test
      </div>
      <button
        aria-label="close"
        class="c6 c7 c8"
        color="primary"
        size="24"
        title="close"
      >
        <div>
          <svg
            aria-hidden="true"
            class="c9 c10"
            fill="currentcolor"
            focusable="false"
            height="24"
            role="img"
            tabindex="-1"
            viewBox="0 0 24 24"
            width="24"
          >
            <path
              d="M19 6.4L17.6 5 12 10.6 6.4 5 5 6.4l5.6 5.6L5 17.6 6.4 19l5.6-5.6 5.6 5.6 1.4-1.4-5.6-5.6L19 6.4z"
            />
          </svg>
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`SmallModalHeader render 1`] = `
.c1 {
  background-color: #0a0;
  color: #fff;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c2 {
  height: 8px;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1 c2"
      color="secondary"
    />
  </div>
</div>
`;
